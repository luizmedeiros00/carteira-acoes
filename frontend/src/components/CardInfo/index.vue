<template>
  <div class="flex flex-col gap-2 p-2 border-l-2">
    <h6 class="text-[10px] font-medium text-gray-400 uppercase">{{ props.label }}</h6>
    <h4 class="text-[14px] font-medium text-primary-500">{{ props.value }}</h4>
  </div>
</template>
<script setup lang="ts">
  interface Props {
    label: string
    value: string
  }

  const props = defineProps<Props>()
</script>
