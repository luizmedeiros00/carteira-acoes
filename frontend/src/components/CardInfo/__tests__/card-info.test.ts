import CardInfo from '../index.vue'

import { shallowMount, VueWrapper } from '@vue/test-utils'

describe('Card Info', () => {

  type Instance = InstanceType<typeof CardInfo>
  let mountFunction: (options?: object) => VueWrapper<Instance>

  beforeEach(() => {
    mountFunction = (options = {}) => {
      return shallowMount(CardInfo, {
        ...options,
      })
    }
  })


  test('should render component and snapshot', () => {
    const wrapper = mountFunction({
      props: {
        label: 'Label test',
        value: 100
      }
    })

    expect(wrapper.html()).toMatchSnapshot()

  })

  test('should render label and value props', () => {
    const wrapper = shallowMount(CardInfo, {
      props: {
        label: 'Label test',
        value: 100
      }
    })

    expect(wrapper.find('h6').text()).toBe('Label test')
    expect(wrapper.find('h4').text()).toBe('100')
  })

})