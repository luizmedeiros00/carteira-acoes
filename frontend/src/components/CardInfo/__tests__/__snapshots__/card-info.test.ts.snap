// Vitest Snapshot v1

exports[`Card Info > should render component and snapshot 1`] = `
"<div class=\\"flex flex-col gap-2 p-2 border-l-2\\">
  <h6 class=\\"text-[10px] font-medium text-gray-400 uppercase\\">Label test</h6>
  <h4 class=\\"text-[14px] font-medium text-primary-500\\">100</h4>
</div>"
`;
